### PR TITLE
[Done] Translate Concurrent Mode API Reference

### PR DESCRIPTION
**Pull request by mohamedsgap**
Started working on 'Concurrent Mode API Reference'

### CODE REVIEW

**Comment by mohamedsgap** [comment]
Need your review guys! @Aissaoui-Ahmed @3imed-jaberi sorry for being late, just finished it.

**Comment by mohamedsgap** [comment]
any updates ??? @Aissaoui-Ahmed @3imed-jaberi

**Comment by asantarissy** [comment]
> any updates ??? @Aissaoui-Ahmed @3imed-jaberi

Good job @mohamedsgap, could you please pull the master branch to make sure that your edits are up to date.

```
> git remote add upstream https://github.com/reactjs/ar.reactjs.org.git
> git checkout doc-concurrent-mode-reference
> git pull upstream master
```

**Comment by mohamedsgap** [comment]
@asantarissy  I think there is a new updates on doc so I gonna translate the new version of this file :)

**Comment by mohamedsgap** [comment]
Finished Translating `Concurrent Mode API Reference`, I need your review guys! @asantarissy @3imed-jaberi @Aissaoui-Ahmed

**Comment by mohamedsgap** [comment]
Finished Translating Concurrent Mode API Reference, I need your review guys! @asantarissy @3imed-jaberi @Aissaoui-Ahmed

File: content/docs/concurrent-mode-reference.md [hunk 1]
```diff
@@ -153,39 +152,38 @@ function App() {
...
 
 ```js
 const deferredValue = useDeferredValue(value, { timeoutMs: 2000 });
 ```
 
-Returns a deferred version of the value that may "lag behind" it for at most `timeoutMs`.
+لعرض نسخة مؤجلة من القيمة التي قد "تتخلف" عنها على الأكثر لـ `timeoutMs`.
 
-This is commonly used to keep the interface responsive when you have something that renders immediately based on user input and something that needs to wait for a data fetch.
+يُستخدم هذا بشكل شائع للحفاظ على استجابة الواجهة عندما يكون لديك شيء يتم تصييره فورًا بناءً على إدخال المستخدم وشيء يحتاج إلى انتظار جلب البيانات.
```
Comment (Fcmam5) [inline content/docs/concurrent-mode-reference.md:179]: I'm not 100% sure about this one, but what do you think of this:

```suggestion
يُستخدم هذا بشكل شائع للحفاظ على استجابة الواجهة عندما يكون لديك شيء يتم تصييره فورًا بناءً على مُدخلات المستخدم وشيء يحتاج إلى انتظار جلب البيانات.
```

File: content/docs/concurrent-mode-reference.md [hunk 2]
```diff
@@ -199,16 +197,16 @@ function App() {
...
 
 ```js
 const SUSPENSE_CONFIG = { timeoutMs: 2000 };
 ```
 
-`useDeferredValue` accepts an **optional Suspense Config** with a `timeoutMs`. This timeout (in milliseconds) tells React how long the deferred value is allowed to lag behind.
+يقبل `useDeferredValue` **Suspense Config  اختياري** مع  `timeoutMs`. هذه المهلة (بالمللي ثانية) تخبرنا React عن المدة التي يُسمح فيها بالقيمة المؤجلة بالتخلف.
 
-React will always try to use a shorter lag when network and device allows it.
+سيحاول React دائمًا استخدام تأخرًا أقصر عندما تسمح به الشبكة والجهاز.
```
Comment (Fcmam5) [inline content/docs/concurrent-mode-reference.md:212]: The word `تأخرًا` is a مضاف إليه is it must be:

```suggestion
سيحاول React دائمًا استخدام تأخرٍ أقصر عندما تسمح به الشبكة والجهاز.
```

Also, I'd go for using `تَبَاطُؤ` as a translation for it (ref: https://www.arabeyes.org/techdict:Lag):

```suggestion
سيحاول React دائمًا استخدام تَبَاطُؤٍ أقصر عندما تسمح به الشبكة والجهاز.
```

File: content/docs/concurrent-mode-reference.md [hunk 3]
```diff
@@ -153,39 +152,38 @@ function App() {
 }
 ```
 
-In this code, we've wrapped our data fetching with `startTransition`. This allows us to start fetching the profile data right away, while deferring the render of the next profile page and its associated `Spinner` for 2 seconds (the time shown in `timeoutMs`).
+في هذه الشفرة، قمنا بلف جلب بياناتنا بـ `startTransition`. يتيح لنا ذلك البدء في جلب بيانات الملف الشخصي فورًا، مع تأجيل عرض صفحة الملف الشخصي التالية و `Spinner` المرتبط بها لمدة ثانيتين (الوقت الموضح في `timeoutMs` ).
```
Comment (Aissaoui-Ahmed) [inline content/docs/concurrent-mode-reference.md:155]:
```suggestion
في هذه الشفرة، قمنا بلف جلب بياناتنا بـ `startTransition`. يتيح لنا ذلك البدء في جلب بيانات الملف الشخصي فورًا، مع تأجيل عرض صفحة الملف الشخصي التالية و `Spinner` المرتبط بها لمدة ثانيتين (الوقت الموضح في `timeoutMs`).
```

File: content/docs/concurrent-mode-reference.md [hunk 4]
```diff
@@ -153,39 +152,38 @@ function App() {
...
 
-The `isPending` boolean lets React know that our component is transitioning, so we are able to let the user know this by showing some loading text on the previous profile page.
+تتيح القيمة المنطقية `isPending` لـ React معرفة أن المكون الخاص بنا يتحول، لذلك يمكننا إخبار المستخدم بذلك من خلال عرض نص التحميل على صفحة الملف الشخصي السابقة.
 
-**For an in-depth look at transitions, you can read [Concurrent UI Patterns](/docs/concurrent-mode-patterns.html#transitions).**
+**لإلقاء نظرة متعمقة على التحولات، يمكنك قراءة [Concurrent UI Patterns](/docs/concurrent-mode-patterns.html#transitions).**
```
Comment (Aissaoui-Ahmed) [inline content/docs/concurrent-mode-reference.md:159]:
```suggestion
**لإلقاء نظرة متعمقة على التحولات، يمكنك قراءة [أنماط و أساليب واجهة المستخدم](/docs/concurrent-mode-patterns.html#transitions).**
```

File: content/docs/concurrent-mode-reference.md [hunk 5]
```diff
@@ -153,39 +152,38 @@ function App() {
...
 #### useTransition Config {#usetransition-config}
 
 ```js
 const SUSPENSE_CONFIG = { timeoutMs: 2000 };
 ```
 
-`useTransition` accepts an **optional Suspense Config** with a `timeoutMs`. This timeout (in milliseconds) tells React how long to wait before showing the next state (the new Profile Page in the above example).
-
-**Note: We recommend that you share Suspense Config between different modules.**
+يقبل `useTransition` **Suspense Config  اختياري** مع `timeoutMs`. هذه المهلة (بالمللي ثانية) تخبر React  إلى متى الانتظار قبل عرض الحالة التالية (صفحة الملف الشخصي الجديدة في المثال أعلاه).
```
Comment (Aissaoui-Ahmed) [inline content/docs/concurrent-mode-reference.md:167]:
```suggestion
يقبل `useTransition` **Suspense Config اختياري** مع `timeoutMs`. هذه المهلة (بالمللي ثانية) تخبر React إلى متى الانتظار قبل عرض الحالة التالية (صفحة الملف الشخصي الجديدة في المثال أعلاه).
```

File: content/docs/concurrent-mode-reference.md [hunk 6]
```diff
@@ -199,16 +197,16 @@ function App() {
  }
 ```
 
-This allows us to start showing the new text for the `input` immediately, which allows the webpage to feel responsive. Meanwhile, `MySlowList` "lags behind" for up to 2 seconds according to the `timeoutMs` before updating, allowing it to render with the current text in the background.
+يتيح لنا ذلك البدء في عرض النص الجديد لـ `الإدخال` على الفور، مما يسمح لصفحة الويب بالشعور بالاستجابة. في هذه الأثناء، `MySlowList` "متخلفة" لمدة تصل إلى ثانيتين وفقًا لـ `timeoutMs`  قبل التحديث، مما يتيح تصييرها مع النص الحالي في الخلفية.
```
Comment (Aissaoui-Ahmed) [inline content/docs/concurrent-mode-reference.md:200]:
```suggestion
يتيح لنا ذلك البدء في عرض النص الجديد لـ `الإدخال` على الفور، مما يسمح لصفحة الويب بالشعور بالاستجابة. في هذه الأثناء، `MySlowList` "متخلفة" لمدة تصل إلى ثانيتين وفقًا لـ `timeoutMs` قبل التحديث، مما يتيح تصييرها مع النص الحالي في الخلفية.
```

File: content/docs/concurrent-mode-reference.md [hunk 7]
```diff
@@ -199,16 +197,16 @@ function App() {
...
+**لإلقاء نظرة متعمقة على تأجيل القيم، يمكنك قراءة [Concurrent UI Patterns](/docs/concurrent-mode-patterns.html#deferring-a-value).**
 
 #### useDeferredValue Config {#usedeferredvalue-config}
 
 ```js
 const SUSPENSE_CONFIG = { timeoutMs: 2000 };
 ```
 
-`useDeferredValue` accepts an **optional Suspense Config** with a `timeoutMs`. This timeout (in milliseconds) tells React how long the deferred value is allowed to lag behind.
+يقبل `useDeferredValue` **Suspense Config  اختياري** مع  `timeoutMs`. هذه المهلة (بالمللي ثانية) تخبرنا React عن المدة التي يُسمح فيها بالقيمة المؤجلة بالتخلف.
```
Comment (Aissaoui-Ahmed) [inline content/docs/concurrent-mode-reference.md:210]:
```suggestion
يقبل `useDeferredValue` **Suspense Config اختياري** مع `timeoutMs`. هذه المهلة (بالمللي ثانية) تخبرنا React عن المدة التي يُسمح فيها بالقيمة المؤجلة بالتخلف.
```

File: content/docs/concurrent-mode-reference.md [hunk 8]
```diff
@@ -70,13 +69,13 @@ Blocking Mode only contains a small subset of Concurrent Mode features and is in
...
-`Suspense` lets your components "wait" for something before they can render, showing a fallback while waiting.
+`Suspense` يتيح لمكوناتك "الانتظار" لشيء ما قبل أن تتمكن من التصيير، مع إظهار احتياطي أثناء الانتظار.
 
-In this example, `ProfileDetails` is waiting for an asynchronous API call to fetch some data. While we wait for `ProfileDetails` and `ProfilePhoto`, we will show the `Loading...` fallback instead. It is important to note that until all children inside `<Suspense>` has loaded, we will continue to show the fallback.
+في هذا المثال، ينتظر `ProfileDetails`  اتصال API غير متزامن لجلب بعض البيانات. أثناء انتظار ` ProfileDetails` و `ProfilePhoto`، سنعرض الخيار `جاري التحميل ...`  بدلاً من ذلك. من المهم ملاحظة أنه حتى يتم تحميل جميع الأطفال داخل `<Suspense>`، سنستمر في إظهار الاحتياط.
 
-`Suspense` takes two props:
-* **fallback** takes a loading indicator. The fallback is shown until all of the children of the `Suspense` component have finished rendering.
-* **unstable_avoidThisFallback** takes a boolean. It tells React whether to "skip" revealing this boundary during the initial load. This API will likely be removed in a future release.
+`Suspense` يأخذ two props:
```
Comment (Aissaoui-Ahmed) [inline content/docs/concurrent-mode-reference.md:76]:
```suggestion
`Suspense` يأخذ اثنين props:
```

File: content/docs/concurrent-mode-reference.md [hunk 9]
```diff
@@ -95,19 +94,19 @@ In this example, `ProfileDetails` is waiting for an asynchronous API call to fet
...
+عندما تحتاج مكونات متعددة إلى جلب البيانات، قد تصل هذه البيانات بترتيب غير متوقع. ومع ذلك، إذا قمت بلف هذه العناصر في `SuspenseList`، فلن تظهر React عنصرًا في القائمة حتى يتم عرض العناصر السابقة (هذا السلوك قابل للتعديل).
 
-`SuspenseList` takes two props:
-* **revealOrder (forwards, backwards, together)** defines the order in which the `SuspenseList` children should be revealed.
-  * `together` reveals *all* of them when they're ready instead of one by one.
-* **tail (collapsed, hidden)** dictates how unloaded items in a `SuspenseList` is shown. 
-    * By default, `SuspenseList` will show all fallbacks in the list.
-    * `collapsed` shows only the next fallback in the list.
-    * `hidden` doesn't show any unloaded items.
+`SuspenseList` يأخذ خاصيَتَيْن:
```
Comment (Aissaoui-Ahmed) [inline content/docs/concurrent-mode-reference.md:101]:
```suggestion
`SuspenseList` يأخذ اثنين props:
```

**Comment by mohamedsgap** [comment]
It's ready to be merged now @Aissaoui-Ahmed  @Fcmam5 @3imed-jaberi

**Comment by mohamedsgap** [comment]
@asantarissy Just merge it, mate 👍

File: content/docs/concurrent-mode-reference.md [hunk 10]
```diff
@@ -95,19 +94,19 @@ In this example, `ProfileDetails` is waiting for an asynchronous API call to fet
...
+`SuspenseList` يأخذ اثنين props:
+* **revealOrder (forwards, backwards, together)** يحدد الترتيب الذي فيه `SuspenseList` يجب الكشف عن الأطفال.
+  * `together` تكشف *الكل* عندما يكونون على استعداد بدلاً من واحدة تِلوَ الأخرى.
+* **tail (collapsed, hidden)** يملي كيف يتم عرض العناصر التي تم تحميلها فى `SuspenseList`.
+    * بشكل افتراضي, `SuspenseList` سيظهر جميع الاحتياطيات في القائمة.
+    * `collapsed` يظهر فقط الاحتياطية التالية في القائمة.
+    * `hidden` لا تظهر أي عناصر غير محمل.
 
-Note that `SuspenseList` only operates on the closest `Suspense` and `SuspenseList` components below it. It does not search for boundaries deeper than one level. However, it is possible to nest multiple `SuspenseList` components in each other to build grids.
+لاحظ أن `SuspenseList`  تعمل فقط على أقرب مكونات `Suspense` و مكونات `SuspenseList`  تحتها. لا يبحث عن حدود أعمق من مستوى واحد. ومع ذلك، فمن الممكن تداخل مكونات `SuspenseList` متعددة في بعضها البعض لبناء شبكات.
```
Comment (3imed-jaberi) [inline content/docs/concurrent-mode-reference.md:109]:
```suggestion
لاحظ أن `SuspenseList`  تعمل فقط على أقرب مكونات `Suspense` و مكونات `SuspenseList`  تحتها. لا يبحث عن حدود أعمق من مستوى واحد. ومع ذلك، فمن الممكن تداخل مكونات `SuspenseList` متعددة في بعضها البعض لبناء شبكات (grids).
```

File: content/docs/concurrent-mode-reference.md [hunk 11]
```diff
@@ -153,39 +152,38 @@ function App() {
...
 #### useTransition Config {#usetransition-config}
 
 ```js
 const SUSPENSE_CONFIG = { timeoutMs: 2000 };
 ```
 
-`useTransition` accepts an **optional Suspense Config** with a `timeoutMs`. This timeout (in milliseconds) tells React how long to wait before showing the next state (the new Profile Page in the above example).
-
-**Note: We recommend that you share Suspense Config between different modules.**
+يقبل `useTransition` **Suspense Config اختياري** مع `timeoutMs`. هذه المهلة (بالمللي ثانية) تخبر React إلى متى الانتظار قبل عرض الحالة التالية (صفحة الملف الشخصي الجديدة في المثال أعلاه).
```
Comment (3imed-jaberi) [inline content/docs/concurrent-mode-reference.md:167]: can you improve more !!!

**Comment by 3imed-jaberi** [comment]
@mohamedsgap, plz don't mark the suggestion as resolved with out any comment ... It's great to discuss !!

**Pull request merged**